src/common/chat-participants/powerpages/commands/create-site/Nl2PageService.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+import { ITelemetry } from "../../../../OneDSLoggerTelemetry/telemetry/ITelemetry";
+import { NL2PAGE_REQUEST_FAILED, VSCODE_EXTENSION_NL2PAGE_REQUEST_FAILED, VSCODE_EXTENSION_NL2PAGE_REQUEST_SUCCESS } from "../../PowerPagesChatParticipantConstants";
+
+export async function getNL2PageData(aibEndpoint: string, aibToken: string, userPrompt: string, siteName: string, sitePagesList: string[], sessionId: string, telemetry: ITelemetry) {
+    const constructRequestBody = (pageType: string) => ({
+        "crossGeoOptions": {
+            "enableCrossGeoCall": true
+        },
+        "question": `${userPrompt} - ${pageType} page`,
+        "context": {
+            "shouldCheckBlockList": false,
+            "sessionId": sessionId,
+            "scenario": "NL2Page",
+            "subScenario": "GenerateNewPage",
+            "version": "V1",
+            "information": {
+                "scope": "Page",
+                "includeImages": true,
+                "pageType": pageType === 'FAQ' ? 'FAQ' : 'Home',
+                "title": siteName,
+                "pageName": pageType,
+                "colorNumber": 7, // Add a function to get a random number
+                "shuffleImages": false,
+                "exampleNumber": 2 // Add a function to get a random number
+            }
+        }
+    });
+
+    const requests = sitePagesList.map(async pageType => {
+        const requestBody = constructRequestBody(pageType);
+
+        const requestInit: RequestInit = {
+            method: "POST",
+            headers: {
+                "Content-Type": "application/json",
```

Comment:
we have a common util function for getHeader - lets use that - already used in PPAPI service